swap/swap.go

```diff
@@ -118,14 +131,15 @@ func swapRotatingFileHandler(logdir string) (log.Handler, error) {
 }
 
 // new - swap constructor without integrity check
-func new(stateStore state.Store, owner *Owner, backend contract.Backend, params *Params) *Swap {
+func new(stateStore state.Store, owner *Owner, backend contract.Backend, params *Params, logger log.Logger) *Swap {
```

Comment:
This is not from this PR, but we need to rename this function, it creates a conflict with the builtin `new` keyword of `go` if it would  be used inside this package.

Call it `newInstance` or something else you prefer. You can also create a different PR for this if you prefer.

Reply:
pushed this change and added it to the PR description. thanks